streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -393,7 +393,7 @@ public static StreamThread create(final TopologyMetadata topologyMetadata,
                                       final Runnable shutdownErrorHook,
                                       final BiConsumer<Throwable, Boolean> streamsUncaughtExceptionHandler) {
 
-        final boolean stateUpdaterEnabled = InternalConfig.stateUpdaterEnabled(config.originals());
```

Comment:
This class is more complex. Might be better to split into a different PR, as it required more cleanup. Doing a partial cleanup for a single class, is not a "clean cut" and make reviewing more complex.